diff --git a/python/setup.py b/python/setup.py
index 22d2878..9b5d7ee 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -35,8 +35,6 @@ root = pathlib.Path(__name__).parent
 # cuda arch check for fp8 at the moment.
 for cuda_arch_flags in torch_cpp_ext._get_cuda_arch_flags():
     arch = int(re.search("compute_\d+", cuda_arch_flags).group()[-2:])
-    if arch < 75:
-        raise RuntimeError("FlashInfer requires sm75+")
 
 enable_bf16 = os.environ.get("FLASHINFER_ENABLE_BF16", "1") == "1"
 enable_fp8 = os.environ.get("FLASHINFER_ENABLE_FP8", "1") == "1"
